openstack/blockstorage/v2/volumes/requests_test.go

```diff
@@ -0,0 +1,197 @@
+package volumes
+
+import (
+	"testing"
+
+	"github.com/rackspace/gophercloud/pagination"
+	th "github.com/rackspace/gophercloud/testhelper"
+	"github.com/rackspace/gophercloud/testhelper/client"
+)
+
+func TestList(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+
+	MockListResponse(t)
+
+	count := 0
+
+	List(client.ServiceClient(), &ListOpts{}).EachPage(func(page pagination.Page) (bool, error) {
+		count++
+		actual, err := ExtractVolumes(page)
+		if err != nil {
+			t.Errorf("Failed to extract volumes: %v", err)
+			return false, err
+		}
+
+		expected := []Volume{
+			{
+				ID:                        "289da7f8-6440-407c-9fb4-7db01ec49164",
+				Name:                      "vol-001",
+				Attachments:               []map[string]interface{}{},
```

Comment:
Optional, but I think it would be beneficial to have some real attachment data here so tests can verify if it is correctly being parsed.